frontend/src/components/common/UploadedFilesList.js

```diff
@@ -0,0 +1,113 @@
+import React from "react";
+import moment from "moment";
+import PropTypes from "prop-types";
+import { Col, Row, Popconfirm, Icon } from "antd";
+import { bindActionCreators } from "redux";
+import { connect } from "react-redux";
+
+import CustomPropTypes from "@/customPropTypes";
+import {
+  removeMineDocumentFromExpectedDocument,
+  fetchMineRecordById,
+} from "@/actionCreators/mineActionCreator";
+import { getMines } from "@/selectors/mineSelectors";
+import { getExpectedDocumentStatusOptions } from "@/selectors/staticContentSelectors";
+
+const propTypes = {
+  selectedDocId: PropTypes.string.isRequired,
+  mineId: PropTypes.string.isRequired,
+  removeMineDocumentFromExpectedDocument: PropTypes.func.isRequired,
+  fetchMineRecordById: PropTypes.func.isRequired,
+  mines: PropTypes.objectOf(CustomPropTypes.mine).isRequired,
+  expectedDocumentStatusOptions: PropTypes.arrayOf(CustomPropTypes.dropdownListItem).isRequired,
+};
+
+export class UploadedFilesList extends React.Component {
+  unlinkFile = (event, mineDocumentGuid, expDocumentGuid) => {
+    event.preventDefault();
+    this.props
+      .removeMineDocumentFromExpectedDocument(mineDocumentGuid, expDocumentGuid)
+      .then(this.props.fetchMineRecordById(this.props.mineId));
+  };
+
+  canUnlink = (selectedDoc) => {
+    const docStatus = this.props.expectedDocumentStatusOptions.find(
+      (status) => status.value === selectedDoc.exp_document_status_guid
+    );
+
+    if (moment(selectedDoc.due_date, "YYYY-MM-DD") > moment()) {
+      if (docStatus.label === "Not Received" || docStatus.label === "Received / Pending Review") {
+        return true;
```

Comment:
Returning a boolean inside of a conditional is redundant. You can just return the condition instead 👍 

In this case, it'll be the two conditions, joined. For readability, it may make sense to give them helpful names and then just `return check1 && check2` (with the helpful names instead of check#)

This also means you don't need the second `return` statement

Reply:
changed to a single return.